derive/src/lib.rs

```diff
@@ -294,6 +302,23 @@ impl TypeInfoImpl {
     }
 }
 
+// /// Find the name given to the `scale-info` crate in the context we run in. If scale-info is not
+// /// among the dependencies then we must be deriving types for the scale-info crate itself, in which
+// /// case we need to refer to it using the reserved word "crate", so the object paths keep working.
+// fn scale_info_crate_path() -> Result<syn::Path> {
+//     use proc_macro_crate::FoundCrate;
+//     const SCALE_INFO_CRATE_NAME: &str = "scale-info";
+
+//     let crate_ident = match proc_macro_crate::crate_name(SCALE_INFO_CRATE_NAME) {
+//         Ok(FoundCrate::Itself) => parse_quote! { crate },
+//         Ok(FoundCrate::Name(name)) => {
+//             let crate_ident = Ident::new(&name, Span::call_site());
+//             parse_quote! { :: #crate_ident }
+//         }
+//         Err(e) => return Err(syn::Error::new(Span::call_site(), e)),
+//     };
+//     Ok(crate_ident)
+
```

Comment:
Wasn't entirely sure which version of this function to use so went with master. If that's fine we can delete this commented out version.